server/src/tests/url.test.js

```diff
@@ -0,0 +1,188 @@
+
+      expect(res.status).to.equal(StatusCodes.OK);
+      expect(res.body).to.have.length.above(0);
+    });
+
+    after(async () => {
+      await urlCreateeService.addNewUrl(
+        urls.REAL_URL.originalUrl,
+        urls.REAL_URL.shorterUrl
+      );
+    });
+
+    it("Should return a 404 status code if urls not found in db", async () => {
+      const beforeDeletingRes = await request(app).get("/url/all");
+      expect(beforeDeletingRes.body).to.have.length.above(0);
+
+      await urlDeleteService.deleteAllUrls();
+
+      const afterDeletingRes = await request(app).get("/url/all");
+
+      expect(afterDeletingRes.status).to.equal(StatusCodes.NOT_FOUND);
+    });
+  });
+
+  describe("Gets entity by shorter url", () => {
+    it("Should return a 302 status code and redirect to the original url with the shorter url", async () => {
+      const res = await request(app).get(`/url/${urls.REAL_URL.shorterUrl}`);
+      expect(res.status).to.equal(StatusCodes.MOVED_TEMPORARILY);
+      expect(res.header["location"]).to.equal(urls.REAL_URL.originalUrl);
+    });
+
+    it("Should return a 404 status code if shorter url not found", async () => {
+      const res = await request(app).get(`/url/${urls.MOCKED_URL.shorterUrl}`);
+      expect(res.status).to.equal(StatusCodes.NOT_FOUND);
+    });
+  });
+
+  describe("Gets entities by starts with a specific string", () => {
+    it("Should return a 200 status code and all the entities that starts with a specific string", async () => {
+      const res = await request(app).get(
+        `/url/startWith/${urls.REAL_URL.shorterUrl[0]}`
+      );
+
+      expect(res.status).to.equal(StatusCodes.OK);
+      expect(res.body).to.have.lengthOf(1);
+    });
+
+    it("Should return a 404 status code if shorter urls that starts with a specific string not found", async () => {
+      const res = await request(app).get("/url/startWith/!");
+      expect(res.status).to.equal(StatusCodes.NOT_FOUND);
+    });
+  });
+
+  describe("Gets entities by contains a specific string", () => {
+    it("Should return a 200 status code and all the entities that contains a specific string", async () => {
+      const res = await request(app).get(
+        `/url/contains/${urls.REAL_URL.shorterUrl[1]}`
+      );
+
+      expect(res.status).to.equal(StatusCodes.OK);
+      expect(res.body).to.have.lengthOf(1);
+    });
+
+    it("Should return a 404 status code if shorter urls that contains a specific string not found", async () => {
+      const res = await request(app).get("/url/contains/!");
+      expect(res.status).to.equal(StatusCodes.NOT_FOUND);
+    });
+  });
+
+  describe("Gets entities that don't contain a specific string", () => {
+    it("Should return a 200 status code and all the entities that don't contain a specific string", async () => {
+      const res = await request(app).get("/url/notContains/!");
+
+      expect(res.status).to.equal(StatusCodes.OK);
+      expect(res.body).to.have.lengthOf(1);
```

Comment:
I'd write the 1 in a const, it's hard to understand what it means when it's hard coded